assets/scripts/store/actions/street.js

```diff
@@ -76,9 +78,22 @@ export const clearSegmentsAction = () => {
   }
 }
 
+export const updateStreetSegmentCapacity = (index, value) => {
+  return async (dispatch, getState) => {
+    await dispatch(changeSegmentProperties(index, { capacity: value }))
+    saveStreetToServerIfNecessary()
+  }
+}
 export const updateStreetAnalytics = (isVisible) => {
   return async (dispatch, getState) => {
-    await dispatch(updateAnalytics(isVisible))
+    await dispatch(updateShowAnalytics(isVisible))
+    await dispatch(segmentsChanged())
+  }
+}
+
+export const setAnalyticsSource = (source) => {
+  return async (dispatch, getState) => {
+    await dispatch(updateAnalyticsSource(source))
```

Comment:
See above comments about possibly improving performance by making these changes in one dispatch rather than multiple sequential dispatched actions.

Reply:
Updated this action so it's the only one we need to dispatch 👌